docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,286 @@
+		2. **XN** - Execute Never
+
+- For configuring an MPU region, we need to set below MPU registers:
+	1. **MPU_RNR** : MPU Region Number register
+	2. **MPU_RBAR** : MPU Region Base Address register
+	3. **MPU_RASR** : MPU Region Attribute and Size Register
+
+- MPU_RNR is a unified identifier to the MPU system. This value should not exceed max number of MPU regions for the processor.
+- MPU_RBAR defines the starting address of a MPU region.
+- MPU_RASR encodes size of MPU region and attributes like read/write access, execute permission etc.
+
+- When configuring MPU regions, take into account that for overlapping memory ranges b/w MPU regions, attributes of highest-numbered MPU region will overwrite those set by lower MPU regions for that address range.
+
+## Prerequisites for MPU in TizenRT
+
+- In TizenRT, support for configuring MPU regions is available only for Protected Build.
+- The following configs need to be enabled for configuring MPU regions(for ARMv7 architecture):
+
+```
+	---------------------------------------------------------------------------------
+	CONFIG FLAG			VALUE		DESCRIPTION
+	---------------------------------------------------------------------------------
+	CONFIG_BUILD_PROTECTED		y		TizenRT Protected build
+	CONFIG_ARMV7M_MPU		y		MPU support
+	CONFIG_ARMV7M_MPU_NREGIONS	16		Number of MPU regions
+	---------------------------------------------------------------------------------
+```
+
+- Similarly, for other architecture versions, corresponding MPU related configs for that architecture should be enabled.
+- In addition to these essential configs, below configs will decide which MPU region is used for what purpose:
+
+```
+	--------------------------------------------------------------------------------------------
+	CONFIG FLAG				TYPE		DESCRIPTION
+	--------------------------------------------------------------------------------------------
+	CONFIG_APP_BINARY_SEPARATION		Boolean		Application binary separation build
+	CONFIG_SUPPORT_COMMON_BINARY		Boolean		Support for Common Binary
+	CONFIG_MPU_STACK_OVERFLOW_PROTECTION	Boolean		MPU Stack Overflow Protection
+	CONFIG_OPTIMIZE_APP_RELOAD_TIME		Boolean		Optimizations for app reload time
+	---------------------------------------------------------------------------------------------
+```
+
+## Functions used and Enumeration for MPU regions
+
+- The following 2 functions are used for configuring MPU regions for User Internal SRAM and User Flash in case of no App-Binary Separation.
+
+```c
+static inline void mpu_userflash(uint32_t region, uintptr_t base, size_t size)
+{
+.
+.
+.
+}
+
+static inline void mpu_userintsram(uint32_t region, uintptr_t base, size_t size)
+{
+.
+.
+.
+}
+```
+
+- In case of App-Binary Separation, below function is used for MPU region configuration for memory ranges in User RAM.
+
+```c
+void mpu_get_register_value(uint32_t \*regs, uint32_t region, uintptr_t base, size_t size, uint8_t readonly, uint8_t execute)
+{
+.
+.
+.
+}
+```
```

Comment:
May be a brief argument description for the above functions can be added.

Reply:
done